effects/antidot_effect.py

```diff
@@ -0,0 +1,38 @@
+import message_text_config as msg
+import user_interaction
+import utils
+from citizens.citizen import Citizen
+from citizens.player import Player
+from game import Game
+
+from effects.effect import Effect
+
+
+class AntidotEffect(Effect):
+    def __init__(self, game: Game, name: str, creator: Citizen) -> None:
+        super().__init__(game, name, creator, 12)
+
+    def _activate_impl(self) -> bool:
+        target_number = utils.read_target_number(
+            self.context, msg.AntidoteMessages.ACTIVATION_CHOOSE_TARGET,
+            self._validate)
+        self.targets.append(self.game.citizens[target_number - 1])
+
+        self.user_interaction.save_active(
+            msg.AntidoteMessages.ACTIVATION_SUCCESS.format(
+                self.targets[0].name))
+
+        return True
+
+    def _resolve_impl(self) -> bool:
+        if self.activation_round == self.game.round_number:
+            if self.targets[0].hp - 1 == 0:
```

Comment:
Also, don't need such a complicated formulas in if statement :)
`self.targets[0].hp - 1`

Reply:
Changed all resolve_impl method